lib/manager/cargo/readme.md

```diff
@@ -0,0 +1,219 @@
+## Overview
+
+#### Name of package manager
+
+[Cargo](https://doc.rust-lang.org/cargo/index.html)
+
+---
+
+#### What language does this support?
+
+Rust
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+No
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+It uses exclusively `Cargo.toml` files.
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+Cargo.toml
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No.
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No.
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+It is possible to have local dependencies, by specifying a file path.
+
+---
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No a single Cargo.toml file specifies a single package.
```

Comment:
Wouldn't Cargo's "workspaces" concept mean we need to treat them as an interlinked group and not just as separate/autonomous packages?

Reply:
> A workspace is a set of packages that share the same Cargo.lock and output directory.

see [reference](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html#creating-a-workspace)

It looks like if there is a workspace, it would only affect `getArtifacts` function. We would have to run `cargo update` inside an actual repository, or at least in a directory with proper file structure.
Instead of running `cargo update` for each `Cargo.toml` separately if the corresponding `Cargo.lock` exists.

Reply:
It means we need to be able to locate the “root” Cargo.toml, which doesn’t necessarily need to be at the root level of the repository, right? But I think for launch you were planning to support only projects with a Cargo.toml at the root anyway?

Reply:
> It means we need to be able to locate the “root” Cargo.toml, which doesn’t necessarily need to be at the root level of the repository, right?

Yes, most likely. But identifying "root" Cargo.toml seems to be an easy problem, since it would contain a `[workspace]` section, see [reference](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html#creating-a-workspace).

Also it looks like `cargo` treats path dependencies similarly to members of a workspace, it keeps a single `Cargo.lock` file in root directory. And in order to run `cargo update` it needs to have `Cargo.toml` files for them (located at correct paths).

Reply:
So do we need to run cargo update only in the root workspace? Or run it in the current directory of whatever toml file is updated?

Reply:
It doesn't matter, you can run `cargo update` from root directory or any subdirectory of the crate,
It must have proper file structure. like:
```
├── Cargo.lock
├── Cargo.toml
├── add-one
│   ├── Cargo.toml
│   └── src
│       └── lib.rs
├── adder
│   ├── Cargo.toml
│   └── src
│       └── main.rs
└── target
```
It would probably be enough to create empty `src/main.rs` files in order to make subcrates valid, like in current version.